scenario/state.py

```diff
@@ -140,6 +140,83 @@ def copy(self) -> "Self":
         return copy.deepcopy(self)
 
 
+@dataclasses.dataclass(frozen=True)
+class CloudCredential:
+    auth_type: str
+    """Authentication type."""
```

Comment:
feels like this one should be an enum, no?

Reply:
This one I'm less sure about. Do we know the possible range of values, or is it cloud-specific? If it's cloud-specific, I'd say we don't want to enum it.

Reply:
[Checked with the Juju team] and there's a [list of possible types in Juju](https://github.com/juju/juju/blob/3.5/cloud/clouds.go#L43). That gets limited down to a subset depending on which cloud it is.

I feel like an enum (with all the possible auth types) makes sense here, but I wouldn't try to e.g. have the consistency checker verify that the auth type matches the cloud type.

Reply:
Thanks for checking in the Juju code. `scenario.CloudAuthType.FOO` it is then.